ntoskrnl/ex/sysinfo.c

```diff
@@ -1303,12 +1303,15 @@ QSI_DEF(SystemHandleInformation)
 
     DPRINT("NtQuerySystemInformation - SystemHandleInformation\n");
 
-    /* Set initial required buffer size */
-    *ReqSize = FIELD_OFFSET(SYSTEM_HANDLE_INFORMATION, Handles);
+    /* Set minimal input buffer size */
+    *ReqSize = sizeof(SYSTEM_HANDLE_INFORMATION);
```

Comment:
What happens on Windows if you call `NtQuerySystemInformation` for that `SystemHandleInformation` class, with the `ReqSize` pointer parameter being NULL?

Reply:
Nothing special: that parameter is optional. (See #3910.)

Reply:
Precisely, and this is why here, if the user sets `ReqSize` to be NULL, now here and in the other places below (and in the original code as well) there would be a NULL pointer dereference.